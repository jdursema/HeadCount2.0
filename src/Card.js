import React from 'react';
import './style/Card.css'
import PropTypes from 'prop-types';

const Card = ({location, data, changeClass, type}) => {
  const years = Object.keys(data)
  return (
    <div className={type}
         onClick={() => changeClass(location)} >
      <h3>{location}</h3>
        <table>
        {
          years.map((year, index) => {
            if(data[year]>=.5){
              return (
                  <tr>
                  <td className='high'>{year}: {data[year]}</td>
                  </tr>
                )
            } else {
              return (
                 <tr>
                  <td className='low'>{year}: {data[year]}</td>
                 </tr>
                )
            }
            
          })
        }
      </table>
    </div>
  )
}

export default Card;

Card.propTypes = {
  location: PropTypes.string,
  data: PropTypes.object
}
