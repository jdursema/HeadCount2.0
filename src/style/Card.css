.card{
  align-items: center;
  background-color: #F9F7F8;
  box-shadow: 5px 7px 6px grey;
  border: 5px solid black;
  border-radius: 40px;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
  margin-bottom: 2%;
  width: 305px;
  padding: 1% 0%;

}

h3 {
  color: #0D334F;
  margin-bottom: 5px;
}

.high {
  color:black;
}

.low {
  color: red;
}

.card.selected {
  border: 5px solid #655F8E;
  background-color: #DDC9D3;
  opacity: 0.6;
}

table {
  margin-bottom: 10px;
}

