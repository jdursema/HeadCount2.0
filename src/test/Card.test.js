import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from '../Card';

describe('Card Test', () => {

  it('should be defined', () => {
    const mockFunc = jest.fn()
    const data = {location: 'COLORADO', data: {2004: 1, 2005: .5}}
    const type = 'card'   
    const card = shallow(<Card location={data.location} data={data.data} changeClass={mockFunc} type={type}/>)
    expect(card).toBeDefined();

  })

  it ('should receive props', ()=>{
  const mockFunc = jest.fn();
  const data = {location: 'COLORADO', data: {2004: 1, 2005: .5}}
  const type = 'card'
  const card = shallow(<Card location={data.location} data={data.data} changeClass={mockFunc} type={type}/>)
  
  expect(card.instance().props.location).toEqual('COLORADO')
  expect(card.instance().props.data).toEqual({2004: 1, 2005: .5})
  expect(card.instance().props.changeClass).toEqual(mockFunc)
  expect(card.instance().props.type).toEqual(type)
  
  })

  it ('should change className after being clicked', () =>{
    const mockFunc = jest.fn();
    const data = {location: 'COLORADO', data: {2004: 1, 2005: .5}}
    const type = 'card'
    const card = shallow(<Card location={data.location} data={data.data} changeClass={mockFunc} type={type}/>)

    expect(mockFunc.mock.calls.length).toEqual(0)

    card.simulate('click')

    expect(mockFunc.mock.calls.length).toEqual(1)
  })

  it ('should have a className of high if the data is over or equal to .5, and low if its under .5', ()=>{
    const mockFunc = jest.fn();
    const data = {location: 'COLORADO', data: {2004: 1, 2005: .5, 2006: .3}}
    const type = 'card'
    const card = shallow(<Card location={data.location} data={data.data} changeClass={mockFunc} type={type}/>)

    expect(card.find('.high').length).toEqual(2)
    expect(card.find('.low').length).toEqual(1)
    
  });

  it('should match snap shot of the card', () => {
    const mockFunc = jest.fn();
    const data = {location: 'COLORADO', data: {2004: 1, 2005: .5}}
    const type = 'card'
    const card = shallow(<Card location={data.location} data={data.data} changeClass={mockFunc} type={type}/>);

    expect(card).toMatchSnapshot();
  })
  

})
