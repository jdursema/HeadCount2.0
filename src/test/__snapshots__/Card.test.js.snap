exports[`Card Test should match snap shot of the card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    changeClass={[Function]}
    data={
        Object {
            "2004": 1,
            "2005": 0.5,
          }
    }
    location="COLORADO"
    type="card" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          COLORADO
</h3>,
        <table>
          <tbody>
                    <tr
                              className="high">
                              <td>
                                        2004
                                        : 
                              </td>
                              <td>
                                        1
                              </td>
                    </tr>
                    <tr
                              className="high">
                              <td>
                                        2005
                                        : 
                              </td>
                              <td>
                                        0.5
                              </td>
                    </tr>
          </tbody>
</table>,
      ],
      "className": "card  COLORADO",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tbody>
            <tr
                        className="high">
                        <td>
                                    2004
                                    : 
                        </td>
                        <td>
                                    1
                        </td>
            </tr>
            <tr
                        className="high">
                        <td>
                                    2005
                                    : 
                        </td>
                        <td>
                                    0.5
                        </td>
            </tr>
</tbody>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr
                className="high">
                <td>
                                2004
                                : 
                </td>
                <td>
                                1
                </td>
</tr>,
              <tr
                className="high">
                <td>
                                2005
                                : 
                </td>
                <td>
                                0.5
                </td>
</tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    2004
                    : 
</td>,
                  <td>
                    1
</td>,
                ],
                "className": "high",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "2004",
                      ": ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "2004",
                    ": ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    2005
                    : 
</td>,
                  <td>
                    0.5
</td>,
                ],
                "className": "high",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "2005",
                      ": ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "2005",
                    ": ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0.5,
                  },
                  "ref": null,
                  "rendered": 0.5,
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            COLORADO
</h3>,
          <table>
            <tbody>
                        <tr
                                    className="high">
                                    <td>
                                                2004
                                                : 
                                    </td>
                                    <td>
                                                1
                                    </td>
                        </tr>
                        <tr
                                    className="high">
                                    <td>
                                                2005
                                                : 
                                    </td>
                                    <td>
                                                0.5
                                    </td>
                        </tr>
            </tbody>
</table>,
        ],
        "className": "card  COLORADO",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tbody>
              <tr
                            className="high">
                            <td>
                                          2004
                                          : 
                            </td>
                            <td>
                                          1
                            </td>
              </tr>
              <tr
                            className="high">
                            <td>
                                          2005
                                          : 
                            </td>
                            <td>
                                          0.5
                            </td>
              </tr>
</tbody>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr
                  className="high">
                  <td>
                                    2004
                                    : 
                  </td>
                  <td>
                                    1
                  </td>
</tr>,
                <tr
                  className="high">
                  <td>
                                    2005
                                    : 
                  </td>
                  <td>
                                    0.5
                  </td>
</tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      2004
                      : 
</td>,
                    <td>
                      1
</td>,
                  ],
                  "className": "high",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "2004",
                        ": ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "2004",
                      ": ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 1,
                    },
                    "ref": null,
                    "rendered": 1,
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      2005
                      : 
</td>,
                    <td>
                      0.5
</td>,
                  ],
                  "className": "high",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "2005",
                        ": ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "2005",
                      ": ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 0.5,
                    },
                    "ref": null,
                    "rendered": 0.5,
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
